src/python/pants/backend/docker/util_rules/docker_build_args.py

```diff
@@ -13,7 +14,25 @@
 
 
 class DockerBuildArgs(KeyValueSequenceUtil):
-    pass
+    """Collection of arguments to pass to a Docker build."""
+
+    def with_overrides(self, overrides: DockerBuildArgs, only_with_value: bool) -> dict[str, str]:
+        """Override the values in this collection.
+
+        :param only_with_value: whether to return only those key-value pairs which have a truthy value
+        """
+        values = {k: overrides.to_dict().get(k, v) for k, v in self.to_dict().items()}
```

Comment:
rather call `overrides.to_dict()` once outside the dict comprehension to avoid calling it for every key in self.